### PR TITLE
Implementing respawn location.

### PR DESCRIPTION
**Pull request by svaningelgem**
# Description
Adding respawn location attribute.

Creating this as a draft because I see there is a `respawn` function available (and tested). But I don't know exactly what `.respawn()` should do: should it go to the respawn location, or to the bed? What happens exactly?

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Code follows existing style.
- [x] Unit tests added (if applicable).

### CODE REVIEW

**Comment by Thorinwasher** [comment]
> should it go to the respawn location, or to the bed? What happens exactly?

I recomend looking into the Paper code for this, as our behavior in mockbukkit is meant to copy that of paper. Anyhow for your specific question; it looks like there is only one variable that defines the respawn location in Paper. The bed respawn location is that location, here's the implementation of one of the methods:
```java
    public Location getBedSpawnLocation() {
        return this.getRespawnLocation();
    }
```
I hope that clears that up for you

**Comment by svaningelgem** [comment]
Thanks for the hint @Thorinwasher! I have now a copy of the papers code on my computer.
The changes are now in this branch.

**Comment by Thorinwasher** [comment]
> The changes are now in this branch.

Good! Now you need to change the formatting of the code to follow the same style as everything else.

**Comment by svaningelgem** [comment]
No `ruff` or `black` or anything like that for java? :'(

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 1]
```diff
@@ -1964,20 +1964,11 @@ public int getExperiencePointsNeededForNextLevel()
...
 	}
 
-
-	@Nullable
 	@Override
-	public Location getBedSpawnLocation()
-	{
-		return bedSpawnLocation;
-	}
+	public @Nullable Location getRespawnLocation() { return this.respawnLocation; }
```
Comment (Thorinwasher) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java]: This here is what I meant reformat the code. Here's an example
```suggestion
	public @Nullable Location getRespawnLocation()
	{ 
		return this.respawnLocation; 
	}
```

Comment (svaningelgem) [reply src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java]: Ok, this should have been reformatted now.

**Review by Thorinwasher — APPROVED** [review]

**Pull request merged**